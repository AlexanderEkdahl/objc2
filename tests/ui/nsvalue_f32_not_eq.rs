//! Test that NSValue has proper bounds on its Eq implementation.

use objc2_foundation::NSValue;

fn needs_eq<T: Eq>() {}

fn main() {
    // Compiles fine
    needs_eq::<NSValue<u32>>();
    // Fails
    needs_eq::<NSValue<f32>>();
}
