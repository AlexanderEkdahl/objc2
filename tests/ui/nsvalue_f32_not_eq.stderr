error[E0277]: the trait bound `f32: Eq` is not satisfied
  --> ui/nsvalue_f32_not_eq.rs:11:5
   |
11 |     needs_eq::<NSValue<f32>>();
   |     ^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Eq` is not implemented for `f32`
   |
   = help: the following other types implement trait `Eq`:
             i128
             i16
             i32
             i64
             i8
             isize
             u128
             u16
           and 4 others
   = note: required because of the requirements on the impl of `Eq` for `NSValue<f32>`
note: required by a bound in `needs_eq`
  --> ui/nsvalue_f32_not_eq.rs:5:16
   |
5  | fn needs_eq<T: Eq>() {}
   |                ^^ required by this bound in `needs_eq`
